crates/assists/src/handlers/extract_struct_from_enum_variant.rs

```diff
@@ -37,9 +37,21 @@ pub(crate) fn extract_struct_from_enum_variant(
         ast::StructKind::Tuple(field_list) => field_list,
         _ => return None,
     };
+    // Grab the tuple field inner type as a string (if it exists and is alone)
+    // to check against structs that may have a differing name from the enum
+    // variant itself
+    let variant_inner = {
+        let mut fields = field_list.fields();
+        let ty = fields.next().and_then(|field| field.ty()).map(|ty| ty.to_string());
+
+        match fields.next() {
+            Some(_) => None,
+            None => ty,
+        }
+    };
```

Comment:
Hm, why do we need this check? I'd expect something much simpler, like "if number of fields == 1, do nothing"

Reply:
I'm not sure if I follow -- I was under the assumption that the fix for the issue would be to check that, for single-element tuple variants, there needs to be an additional check that no types in the parent scope that also have that name, which avoids the false positive shown in the issue. The assist would be valid for extracting out >1 element tuples regardless, therefore it checks to see if there is more than one element in the tuple.

Reply:
I mean, for 

```rust
enum NameClass {
  ConstReference(Definition)
}
```

The assist should not be applicable regardnless of whether `Definition` exists in the scope or not. This can be implemented as a purely syntactic check, I don't think we need to look at `ty`s here.

Reply:
Ah okay, so it's simply not applying the assist to 1-tuple variants. I'll go ahead and rewrite the PR to do that instead :) thanks for the clarification